src/main/resources/templates/webssh.html

```diff
@@ -0,0 +1,129 @@
+                    <button id="disconnect" class="btn btn-default" type="submit" disabled="disabled">Disconnect</button>
+                </div>
+            </form>
+        </div>
+        <div>
+            <br/>
+            <div id="terminal"></div>
+        </div>
+    </div>
+</div>
+<p>&nbsp;</p>
+<p>&nbsp;</p>
+<div th:replace="fragments/footer :: footer"></div>
+<object th:include="fragments/javascript :: javascript" th:remove="tag" />
+<script src="https://cdnjs.cloudflare.com/ajax/libs/sockjs-client/1.1.4/sockjs.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/stomp.js/2.3.3/stomp.min.js"></script>
+<script type="text/javascript" src="static/js/xterm.js" th:src="@{~/js/xterm.js}"></script>
+<script th:inline="javascript">
+    var stompClient = null;
+    var term = new Terminal();
+    var command = "";
+
+    function setConnected(connected) {
+        $('#connect').prop("disabled", connected);
+        $('#disconnect').prop("disabled", !connected);
+    }
+
+    function connect() {
+        /*<![CDATA[*/
+        var username = [[${deterUid}]];
+        var qualified = [[${qualified}]];
+        /*]]>*/
+        var password = $('#password').val();
+        var socket = new SockJS('/websocket');
+        stompClient = Stomp.over(socket);
+        stompClient.debug = null;
+        stompClient.connect(
+            {user: username, qualified: qualified, pass: password},
+            function (frame) {
+                console.log("Connected: " + frame);
+                term.reset();
+                setConnected(true);
+                stompClient.subscribe('/shell/output/' + username + "@" + qualified, function (output) {
+                    showResult(output.body);
+                });
+                term.focus();
+            },
+            function (error) {
+                term.write("Unable to established ssh connection. Password may be invalid.");
+                disconnect();
+            }
+        );
+    }
+
+    function disconnect() {
+        term.blur();
+        if (stompClient !== null) {
+            stompClient.disconnect();
+        }
+        setConnected(false);
+        console.log("Disconnected");
+    }
+
+    function sendKey(ctrlKey, key, keyCode) {
+        stompClient.send("/webssh/input", {}, JSON.stringify({'ctrlKey':ctrlKey,'key':key,'keyCode':keyCode}));
+    }
+
+    function showResult(message) {
+        term.write(message);
+    }
+
+    $(function () {
+        $('form').on('submit', function (e) {
+            e.preventDefault();
+        });
+        $('#connect').click(function () {
+            connect();
+        });
+        $('#disconnect').click(function () {
+            disconnect();
+        });
+
+        term.open(document.getElementById('terminal'));
+        // Log the keyCode of every keyDown event
+        term.textarea.onkeydown = function (e) {
+            console.log("onkeydown: ", e);
```

Comment:
not necessary to log the keys?

Reply:
Oops again. Forgot to remove. I added so many log messages that I don't remember where they are 
liao.